### PR TITLE
catppuccin-kde: unstable-2022-11-26 -> 0.2.2

### PR DESCRIPTION
**Pull request by michaelBelsanti**
###### Description of changes
Updated the theme to the newest tagged revision. The main change is an installation script with accent options. Unfortunately, this script takes user input, so I had to modify it to work with a declarative install. The package allows for any combination of theme flavours and accents to be installed. 

https://github.com/catppuccin/kde/releases/tag/v0.2.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/catppuccin-kde/default.nix [hunk 1]
```diff
@@ -1,24 +1,45 @@
 { lib
 , stdenvNoCC
 , fetchFromGitHub
+, flavour ? [ "mocha" "macchiato" "frappe" "latte" ]
```
Comment (michaelBelsanti) [inline pkgs/data/themes/catppuccin-kde/default.nix:4]: Weird minor problem, this should be named `flavours`, but if I call it that, then trying to refer to it in the installPhase instead refers to the packages `flavours`.

File: pkgs/data/themes/catppuccin-kde/default.nix [hunk 2]
```diff
@@ -1,24 +1,47 @@
 { lib
 , stdenvNoCC
 , fetchFromGitHub
+, flavour ? [ "frappe" ]
```
[the remark this answers is not in the document]
Comment (NickCao) [reply pkgs/data/themes/catppuccin-kde/default.nix:4]: That's how callPackage works.

**Comment by NickCao** [comment]
User inputs can be emulated with `expect`, I think that is more maintainable than recreating the installation logic ourselves.

**Comment by michaelBelsanti** [comment]
The script doesn't entirely recreate the installation logic, it mostly justs replaces the user input with environment variables. A problem with using expect is that the script takes numbers, not flavour and accent names. I'm not sure how I'd convert from the set flavours and accents to the numbers for the input.

**Pull request merged**